test/unit/tasks/deleted_objects_test.rb

```diff
@@ -30,5 +30,40 @@ class DeletedObjectsTest < ActiveSupport::TestCase
       end
       assert_raise(ActiveRecord::RecordNotFound) { object_with_service_owner_destroyed.reload }
     end
+
+    test 'delete_the_stale_object' do
+      deleted_child_and_owner_service_deleted_old = deleted_object_service_but_owner_persisted_old = nil
+      Timecop.freeze((1.week + 1.day).ago) do
+        service = FactoryBot.create(:simple_service)
+        metric = FactoryBot.create(:metric, service: service)
+        DeletedObject.create(object: metric, owner: service)
```

Comment:
IMO, it is easier to understand adding names to these cases, as we do in  https://github.com/3scale/porta/blob/933da0da4b12d1296e4960b83ca60f3f2af07f68/test/unit/deleted_object_test.rb#L48-L88

But this is just my subjective opinion 😄